src/request-data/user-preferences/normalizers.js

```diff
@@ -33,6 +33,21 @@ export class SitePreferenceNormalizer extends PreferenceNormalizer {
   static projectSortMode(val) {
     return ['alphabetical', 'latest', 'newest'].includes(val) ? val : 'latest';
   }
+
+  static outdatedVersionWarningDismissDate(val) {
+    // Frontend to Backend
+    if (val instanceof Date) {
+      return val.toISOString();
+    }
+
+    // Backend to Frontend
+    const isoDateRegex = /^\d{4}-\d{2}-\d{2}([T](\d{2}:\d{2}:\d{2}(\.\d+)?Z)?)?$/;
+    if (typeof (val) === 'string' && isoDateRegex.test(val)) {
+      return new Date(val);
```

Comment:
Could it be possible to always return a string so that the return type is consistent? I know right now it's different for Backend vs. Frontend, but I think it'd be a little clearer if it was always the same.